src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -390,6 +390,121 @@ BOOST_AUTO_TEST_CASE(SimpleLinearLayerTest)
   BOOST_REQUIRE_EQUAL(arma::accu(delta), 0);
 }
 
+/**
+ * Linear module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(LinearLayerWeightInitializationTest)
+{
+  size_t inSize = 10, outSize = 4;
+  Linear<>* linear = new Linear<>(inSize, outSize);
```

Comment:
I know it's only the tests, but in my opinion each pointer should be freed as long as it is no longer needed. So, the app shouldn't have any memory leaks in all possible cases.

Reply:
Perhaps I missed something. Why did you use the pointer? Why didn't you define the object as follows
```c++
Linear<> linear(inSize, outSize);
```